client/state/themes/reducer.js

```diff
@@ -162,6 +165,34 @@ export function themeRequests( state = {}, action ) {
 }
 
 /**
+ * Returns the updated Jetpack site wpcom theme install requests state after an action has been
+ * dispatched. The state reflects a mapping of site ID, theme ID pairing to a
+ * boolean reflecting whether a request for the theme install is in progress.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function installThemeOnJetpackRequests( state = {}, action ) {
+	switch ( action.type ) {
+		case THEME_INSTALL_ON_JETPACK_REQUEST:
+		case THEME_INSTALL_ON_JETPACK_REQUEST_SUCCESS:
+		case THEME_INSTALL_ON_JETPACK_REQUEST_FAILURE:
+			return Object.assign( {}, state, {
+				[ action.siteId ]: Object.assign( {}, state[ action.siteId ], {
+					[ action.themeId ]: THEME_INSTALL_ON_JETPACK_REQUEST === action.type
```

Comment:
As we discussed: we should store full relation site - theme - bool. We rely on selectors to figure out proper answer to different type of questions like for example; is there an install process for site ongoing or what themes are being installed right now.